pkg/maps/ctmap/ctmap.go

```diff
@@ -287,17 +287,27 @@ func purgeCtEntry6(m *Map, key CtKey, entry *CtEntry, natMap *nat.Map) error {
 	}
 
 	t := key.GetTupleKey()
+	tupleType := t.GetFlags()
 
-	if t.GetFlags()&tuple.TUPLE_F_IN != 0 {
+	if tupleType == tuple.TUPLE_F_IN && entry.isDsrEntry() {
+		// To delete NAT entries created by legacy DSR
+		nat.DeleteSwappedMapping6(natMap, t.(*tuple.TupleKey6Global))
+	}
+
+	if tupleType == tuple.TUPLE_F_OUT {
 		if entry.isDsrEntry() {
-			// To delete NAT entries created by legacy DSR
+			// To delete NAT entries created by DSR
 			nat.DeleteSwappedMapping6(natMap, t.(*tuple.TupleKey6Global))
+		} else {
+			// To delete NAT entries created for SNAT
+			nat.DeleteMapping6(natMap, t.(*tuple.TupleKey6Global))
+
 		}
-	} else if t.GetFlags()&tuple.TUPLE_F_OUT == tuple.TUPLE_F_OUT &&
-		entry.isDsrEntry() {
-		// To delete NAT entries created by DSR
-		nat.DeleteSwappedMapping6(natMap, t.(*tuple.TupleKey6Global))
-	} else {
+	}
+
+	// TODO unclear if this is actually needed
```

Comment:
In general its better to cite a github issue. Then people can be pulled in on the issue then depending on the outcome, remove the citation or adjust the code.

Reply:
I just went ahead and did the BPF digging instead :). See the updated patch description, there's no need to support `TUPLE_F_RELATED`.